src/ui/src/builder/useComponentFieldViewModel.spec.ts

```diff
@@ -0,0 +1,111 @@
+import { describe, it, expect } from "vitest";
+import { ref, nextTick } from "vue";
+import { buildMockComponent, buildMockCore } from "@/tests/mocks";
+import { generateBuilderManager } from "./builderManager";
+import {
+	useComponentFieldViewModel,
+	Dependencies,
+} from "./useComponentFieldViewModel";
+import { FieldType, WriterComponentDefinition } from "@/writerTypes";
+
+describe(useComponentFieldViewModel.name, () => {
+	const componentId = "TestComponent";
+	const fieldKey = "TestField";
+
+	let mockCore: ReturnType<typeof buildMockCore>;
+
+	function setupMockDependencies(
+		componentId: string,
+		fieldKey: string,
+	): Dependencies {
+		mockCore = buildMockCore();
+		mockCore.core.addComponent(
+			buildMockComponent({
+				id: componentId,
+				type: FieldType.Text,
+				content: {
+					[fieldKey]: undefined,
+				},
+			}),
+		);
+
+		function getComponentDefinition(): WriterComponentDefinition {
+			return {
+				name: "",
+				description: "",
+				fields: {
+					[fieldKey]: {
+						name: "",
+						type: FieldType.Text,
+						default: "default test",
+					},
+				},
+			};
+		}
+
+		return {
+			wf: {
+				...mockCore.core,
+				getComponentDefinition,
+			},
```

Comment:
It's better to rely on `vi.spyOn`

```suggestion
		vi.spyOn(mockCore.core, "getComponentDefinition").mockReturnValue({
			name: "",
			description: "",
			fields: {
				[fieldKey]: {
					name: "",
					type: FieldType.Text,
					default: "default test",
				},
			},
		});

		return {
			wf: mockCore.core,
```